### PR TITLE
Do not use isinstance to select array module

### PR DESCRIPTION
**Pull request by delta2323**
Generally speaking, `isinstance` should not be used if possible because it is slow in some case. This PR avoid to use it to determine an array module used.

### CODE REVIEW

**Comment by delta2323** [comment]
Just a comment. We already use `isinstance` via `get_array_module` in the same function. So we still cannot completely remove `isinstance` with this PR.

File: chainer/functions/pooling/unpooling_2d.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def forward(self, x):
         xp = cuda.get_array_module(*x)
         col = xp.tile(x[0][:, :, None, None],
                       (1, 1, self.kh, self.kw, 1, 1))
-        if isinstance(x[0], cuda.ndarray):
```
Comment (gwtnb) [inline chainer/functions/pooling/unpooling_2d.py]: `cuda.cupy` is unavailable on environments without a GPU. How about `xp is not numpy`?

Comment (delta2323) [inline chainer/functions/pooling/unpooling_2d.py]: I fixed. I use `xp is numpy` and switch the order for readbility.

**Comment by gwtnb** [comment]
LGTM

**Pull request merged**